feat(test): Include litmus job to perform disk failure on AWS

dargasudarshan opened the pull request:
Signed-off-by: Sudarshan <sudarshan.darga@cloudbyte.com>

<!--  Thanks for sending a pull request!  Here are some tips for you -->

**What this PR does / why we need it**:
- Include litmus job to perform disk failure on AWS
- Include a util to check if the AUT is deployed on cstor-disk-pool
- Perform detach disk operation.
- Verify cstor pool status on performing disk detach/attach operations.
- Include disk recovery in 2 stages, one in test work flow and one in always block because if test failed after detaching the disk always block tasks helps to recover disk so as to not impact any other tests.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

vibhor995 commented on apps/percona/chaos/disk_failure/run_litmus_test.yml:
```diff
@@ -0,0 +1,108 @@
+
+          ## Value of AWS_ACCESS_KEY_ID need to be replaced with original value
+          - name: AWS_ACCESS_KEY_ID
+            value: "accessid"
```
@dargasudarshan  i guess we are following same naming convention for placeholders 

example:
``` name: AWS_ACCESS_KEY_ID
            value: "access_key"

          # Value of AWS_SECRET_ACCESS_KEY need to be replaced with original value
          - name: AWS_SECRET_ACCESS_KEY
            value: "secret_key"

          - name: SUBNET_ID
            value: "subnet_id"
```

dargasudarshan commented:
Litmus ansibletest logs:

```
ansible-playbook 2.7.5
  config file = None
  configured module search path = [u'/root/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/local/lib/python2.7/dist-packages/ansible
  executable location = /usr/local/bin/ansible-playbook
  python version = 2.7.15rc1 (default, Nov 12 2018, 14:31:15) [GCC 7.3.0]
No config file found; using defaults
/etc/ansible/hosts did not meet host_list requirements, check plugin documentation if this is unexpected
/etc/ansible/hosts did not meet script requirements, check plugin documentation if this is unexpected
statically imported: /percona/chaos/disk_failure/test_prerequisites.yml

PLAYBOOK: test.yml *************************************************************
1 plays in ./percona/chaos/disk_failure/test.yml

PLAY [localhost] ***************************************************************
2018-12-17T09:29:59.285323 (delta: 0.034985)         elapsed: 0.034985 ********
===============================================================================

TASK [Gathering Facts] *********************************************************
task path: /percona/chaos/disk_failure/test.yml:2
2018-12-17T09:29:59.293183 (delta: 0.007838)         elapsed: 0.042845 ********
ok: [127.0.0.1]
META: ran handlers

TASK [include_tasks] ***********************************************************
task path: /percona/chaos/disk_failure/test_prerequisites.yml:3
2018-12-17T09:30:02.033240 (delta: 2.740042)         elapsed: 2.782902 ********
included: /common/utils/application_liveness_check.yml for 127.0.0.1

TASK [Get the liveness pods] ***************************************************
task path: /common/utils/application_liveness_check.yml:2
2018-12-17T09:30:02.102911 (delta: 0.069641)         elapsed: 2.852573 ********
changed: [127.0.0.1] => {"changed": true, "cmd": "kubectl get pod -n litmus -l liveness=percona-liveness -o=custom-columns=NAME:\".metadata.name\" --no-headers", "delta": "0:00:00.500352", "end": "2018-12-17 09:30:02.807489", "rc": 0, "start": "2018-12-17 09:30:02.307137", "stderr": "", "stderr_lines": [], "stdout": "percona-liveness-bqpld-492sb", "stdout_lines": ["percona-liveness-bqpld-492sb"]}

TASK [Checking status of liveness pods] ****************************************
task path: /common/utils/application_liveness_check.yml:6
2018-12-17T09:30:02.842345 (delta: 0.739402)         elapsed: 3.592007 ********
changed: [127.0.0.1] => (item=percona-liveness-bqpld-492sb) => {"attempts": 1, "changed": true, "cmd": "kubectl get pods percona-liveness-bqpld-492sb -n litmus -o=custom-columns=NAME:\".status.phase\" --no-headers", "delta": "0:00:00.352758", "end": "2018-12-17 09:30:03.309324", "item": "percona-liveness-bqpld-492sb", "rc": 0, "start": "2018-12-17 09:30:02.956566", "stderr": "", "stderr_lines": [], "stdout": "Running", "stdout_lines": ["Running"]}




TASK [Fetch AWS instance details] **********************************************
task path: /chaoslib/aws_chaos/disk_failure.yml:11
2018-12-17T09:30:26.279277 (delta: 0.49741)         elapsed: 27.028939 ********
included: /common/utils/aws/fetch_aws_details.yml for 127.0.0.1

TASK [Get the AWS instance Id of Instance] *************************************
task path: /common/utils/aws/fetch_aws_details.yml:3
2018-12-17T09:30:26.341649 (delta: 0.062339)         elapsed: 27.091311 *******
changed: [127.0.0.1] => {"changed": true, "cmd": "kubectl get node ip-10-0-1-127.us-west-1.compute.internal -o jsonpath={.spec.providerID} | cut -d '/' -f 5", "delta": "0:00:00.354745", "end": "2018-12-17 09:30:26.804582", "rc": 0, "start": "2018-12-17 09:30:26.449837", "stderr": "", "stderr_lines": [], "stdout": "i-000e0ae90a0c11837", "stdout_lines": ["i-000e0ae90a0c11837"]}

TASK [Get region from app node] ************************************************
task path: /common/utils/aws/fetch_aws_details.yml:9
2018-12-17T09:30:26.840242 (delta: 0.498562)         elapsed: 27.589904 *******
changed: [127.0.0.1] => {"changed": true, "cmd": "kubectl get nodes ip-10-0-1-127.us-west-1.compute.internal -o jsonpath='{.metadata.labels.failure-domain\\.beta\\.kubernetes\\.io\\/region}'", "delta": "0:00:00.351535", "end": "2018-12-17 09:30:27.298976", "rc": 0, "start": "2018-12-17 09:30:26.947441", "stderr": "", "stderr_lines": [], "stdout": "us-west-1", "stdout_lines": ["us-west-1"]}

TASK [set_fact] ****************************************************************
task path: /common/utils/aws/fetch_aws_details.yml:16
2018-12-17T09:30:27.334375 (delta: 0.494081)         elapsed: 28.084037 *******
ok: [127.0.0.1] => {"ansible_facts": {"instance_id": "i-000e0ae90a0c11837", "region": "us-west-1"}, "changed": false}

TASK [Detach the disk from the node] *******************************************
task path: /chaoslib/aws_chaos/disk_failure.yml:17
2018-12-17T09:30:27.386048 (delta: 0.051644)         elapsed: 28.13571 ********
changed: [127.0.0.1] => {"changed": true, "cmd": "aws ec2 detach-volume --volume-id vol-0751c0d3de2f0a39f --region us-west-1 --instance-id i-000e0ae90a0c11837 --force", "delta": "0:00:01.011580", "end": "2018-12-17 09:30:28.505555", "rc": 0, "start": "2018-12-17 09:30:27.493975", "stderr": "", "stderr_lines": [], "stdout": "{\n    \"AttachTime\": \"2018-12-17T09:20:46.000Z\", \n    \"InstanceId\": \"i-000e0ae90a0c11837\", \n    \"VolumeId\": \"vol-0751c0d3de2f0a39f\", \n    \"State\": \"detaching\", \n    \"Device\": \"/dev/xvdb\"\n}", "stdout_lines": ["{", "    \"AttachTime\": \"2018-12-17T09:20:46.000Z\", ", "    \"InstanceId\": \"i-000e0ae90a0c11837\", ", "    \"VolumeId\": \"vol-0751c0d3de2f0a39f\", ", "    \"State\": \"detaching\", ", "    \"Device\": \"/dev/xvdb\"", "}"]}

TASK [Verify if disk is detached successfully] *********************************
task path: /chaoslib/aws_chaos/disk_failure.yml:20
2018-12-17T09:30:28.540870 (delta: 1.154793)         elapsed: 29.290532 *******
changed: [127.0.0.1] => {"attempts": 1, "changed": true, "cmd": "aws ec2 describe-volumes --volume-id vol-0751c0d3de2f0a39f --region us-west-1 | grep -i state", "delta": "0:00:00.873007", "end": "2018-12-17 09:30:29.526056", "rc": 0, "start": "2018-12-17 09:30:28.653049", "stderr": "", "stderr_lines": [], "stdout": "                    \"State\": \"detaching\", \n            \"State\": \"in-use\", ", "stdout_lines": ["                    \"State\": \"detaching\", ", "            \"State\": \"in-use\", "]}


TASK [Check the cstor-pool status] *********************************************
task path: /percona/chaos/disk_failure/test.yml:84
2018-12-17T09:30:29.634829 (delta: 0.034534)         elapsed: 30.384491 *******
FAILED - RETRYING: Check the cstor-pool status (30 retries left).
FAILED - RETRYING: Check the cstor-pool status (29 retries left).
FAILED - RETRYING: Check the cstor-pool status (28 retries left).
FAILED - RETRYING: Check the cstor-pool status (27 retries left).
FAILED - RETRYING: Check the cstor-pool status (26 retries left).
FAILED - RETRYING: Check the cstor-pool status (25 retries left).
FAILED - RETRYING: Check the cstor-pool status (24 retries left).
FAILED - RETRYING: Check the cstor-pool status (23 retries left).
FAILED - RETRYING: Check the cstor-pool status (22 retries left).
FAILED - RETRYING: Check the cstor-pool status (21 retries left).
FAILED - RETRYING: Check the cstor-pool status (20 retries left).
FAILED - RETRYING: Check the cstor-pool status (19 retries left).
FAILED - RETRYING: Check the cstor-pool status (18 retries left).
FAILED - RETRYING: Check the cstor-pool status (17 retries left).
FAILED - RETRYING: Check the cstor-pool status (16 retries left).
FAILED - RETRYING: Check the cstor-pool status (15 retries left).
FAILED - RETRYING: Check the cstor-pool status (14 retries left).
FAILED - RETRYING: Check the cstor-pool status (13 retries left).
FAILED - RETRYING: Check the cstor-pool status (12 retries left).
FAILED - RETRYING: Check the cstor-pool status (11 retries left).
FAILED - RETRYING: Check the cstor-pool status (10 retries left).
FAILED - RETRYING: Check the cstor-pool status (9 retries left).
FAILED - RETRYING: Check the cstor-pool status (8 retries left).
FAILED - RETRYING: Check the cstor-pool status (7 retries left).
FAILED - RETRYING: Check the cstor-pool status (6 retries left).
FAILED - RETRYING: Check the cstor-pool status (5 retries left).
FAILED - RETRYING: Check the cstor-pool status (4 retries left).
FAILED - RETRYING: Check the cstor-pool status (3 retries left).
FAILED - RETRYING: Check the cstor-pool status (2 retries left).
FAILED - RETRYING: Check the cstor-pool status (1 retries left).
fatal: [127.0.0.1]: FAILED! => {"attempts": 30, "changed": true, "cmd": "kubectl exec -it cstor-disk-pool-vmt5-5876bc9c65-tc2z7 -n openebs -c cstor-pool -- bash -c \"zpool status | grep state\"", "delta": "0:00:00.423676", "end": "2018-12-17 09:33:15.986351", "rc": 0, "start": "2018-12-17 09:33:15.562675", "stderr": "Unable to use a TTY - input is not a terminal or the right kind of file", "stderr_lines": ["Unable to use a TTY - input is not a terminal or the right kind of file"], "stdout": " state: ONLINE", "stdout_lines": [" state: ONLINE"]}

TASK [set_fact] ****************************************************************
task path: /percona/chaos/disk_failure/test.yml:162
2018-12-17T09:33:16.025095 (delta: 166.390236)         elapsed: 196.774757 ****
ok: [127.0.0.1] => {"ansible_facts": {"flag": "Fail"}, "changed": false}
```

Currently the test is failing because of product bug, where the state of cstor pool is not getting changes when a disk is failed.
